commcare_connect/reports/queries.py

```diff
@@ -0,0 +1,44 @@
+from django.db.models import DateTimeField, ExpressionWrapper, F, FloatField
+from django.db.models.fields.json import KeyTextTransform
+from django.db.models.functions import Cast, Extract
+
+
+def get_visit_map_queryset(base_queryset):
+    return (
+        base_queryset.annotate(
+            username_connectid=F("form_json__metadata__username"),
+            deliver_unit_name=F("deliver_unit__name"),
+            days_since_opp_start=ExpressionWrapper(
+                Extract(F("visit_date") - F("opportunity__start_date"), "day"), output_field=FloatField()
+            ),
+            timestart_str=KeyTextTransform("timeStart", KeyTextTransform("metadata", F("form_json"))),
```

Comment:
Can these use (https://docs.djangoproject.com/en/5.1/topics/db/queries/#module-django.db.models.fields.json
0)[KT] and not have to nest? Something like `KT("form_json__timeStart")`? In general curious why some of these need nested transforms and some have `F`, which is likely just my lack of familiarity with json field annotations

Reply:
> In general curious why some of these need nested transforms and some have F, which is likely just my lack of familiarity with json field annotations

I mean, the honest answer is because that's what Claude decided to do. But I can dig into this further for consistency if you think it matters. I also don't have any experience with any of this